src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -323,6 +324,17 @@ template <class IdType>
 CSRMatrix SortedCOOToCSR(const COOMatrix &coo) {
   const int64_t N = coo.num_rows;
   const int64_t NNZ = coo.row->shape[0];
+
+  // TODO(Mingbang): This is just a temporary check to ensure that NNZ does not
```

Comment:
such check code is too duplicate, is it possible to extract into a separate function/template?

Reply:
It's just temporary and will be removed later anyway. Is it necessary?

Reply:
yes. it's worthy. and I don't think the final fix for this issue is expected to be ready very soon.

Reply:
and such check is not **so temporary**, I think. Instead, It's required